util/type_traits.h

```diff
@@ -0,0 +1,69 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Common type traits support
+//
+*******************************************************************************/
+
+#ifndef __SYCLCTS_UTIL_TYPE_TRAITS_H
+#define __SYCLCTS_UTIL_TYPE_TRAITS_H
+
+#include <climits>
+#include <type_traits>
+
+namespace {
+
+// Common type traits functions
+template <typename ... T>
+struct contains;
+
+template <typename T>
+struct contains<T> : std::false_type {};
+
+/**
+ * @brief Verify type is within the list of types
+ */
+template <typename T, typename headT, typename ... tailT>
+struct contains<T, headT, tailT...> :
+    std::conditional<std::is_same<T, headT>::value,
+                     std::true_type,
+                     contains<T, tailT...>>::type {};
+
+/**
+ * @brief Verify type has the given number of bits
+ */
+template <typename T, size_t bits>
+struct bits_eq :
+    std::conditional<(sizeof(T) * CHAR_BIT) == bits,
+                     std::true_type,
+                     std::false_type>::type {};
```

Comment:
```suggestion
template <typename T, size_t bits>
using bits_eq = std::bool_constant<(sizeof(T) * CHAR_BIT) == bits>;
```

Prefer writing type traits in terms of aliases wherever possible - it's clearer and helps with compilation times.

Reply:
Great catch, thanks